sycl/include/sycl/detail/common.hpp

```diff
@@ -83,6 +83,30 @@ struct code_location {
   unsigned long MLineNo;
   unsigned long MColumnNo;
 };
+
+// The C++ FE may instrument user calls with code location metadata.
```

Comment:
nit: I don't think FE does anything here, the solution is purely on the side of the library using common builtins.

Reply:
it is FE that either adds the code_location parameter or not